eventtracking/processors/tests/test_whitelist.py

```diff
@@ -0,0 +1,73 @@
+"""Test the whitelist processor"""
+
+from __future__ import absolute_import
+
+from unittest import TestCase
+
+from mock import sentinel
+
+from eventtracking.processors.exceptions import EventEmissionExit
+from eventtracking.processors.whitelist import NameWhitelistProcessor
+
+
+class TestNameWhitelistProcessor(TestCase):
+    """Test the whitelist processor"""
+
+    def test_filtering_out(self):
+        whitelist = NameWhitelistProcessor(whitelist=[sentinel.allowed_event])
+        with self.assertRaises(EventEmissionExit):
+            whitelist({'name': sentinel.not_allowed_event})
+
+    def test_allowed_event(self):
+        whitelist = NameWhitelistProcessor(whitelist=[sentinel.allowed_event])
+        self.assert_event_passed_through(whitelist, {'name': sentinel.allowed_event})
+
+    def assert_event_passed_through(self, whitelist, event):
+        """Assert that the whitelist allowed the event processing to proceed"""
+        self.assertEquals(whitelist(event), event)
+
+    def test_empty_whitelist(self):
+        whitelist = NameWhitelistProcessor(whitelist=[])
+        with self.assertRaises(EventEmissionExit):
+            whitelist({'name': sentinel.not_allowed_event})
+
+    def test_multi_entry_whitelist(self):
+        whitelist = NameWhitelistProcessor(whitelist=[sentinel.allowed_event, sentinel.another_event])
+        with self.assertRaises(EventEmissionExit):
+            whitelist({'name': sentinel.not_allowed_event})
+
+        self.assert_event_passed_through(whitelist, {'name': sentinel.allowed_event})
+        self.assert_event_passed_through(whitelist, {'name': sentinel.another_event})
+
+    def test_no_whitelist_param(self):
+        with self.assert_initialization_fails():
+            NameWhitelistProcessor()
+
+    def assert_initialization_fails(self):
```

Comment:
nit:  No need to change this, but just to note this going forward.  I have a tiny preference -- that when there's a helper method used in several tests, I find it easier to have the helper first, then the tests.  Or vice versa.  But I find it more confusing to have the helper after the first and before the remaining tests.  (Same thing with 'assert_properly_configured'.)